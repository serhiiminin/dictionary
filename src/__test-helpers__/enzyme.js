import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  })
}

export { shallowWithTheme, mountWithTheme }
