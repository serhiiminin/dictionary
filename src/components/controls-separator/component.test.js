import React from 'react';
import { shallow } from 'enzyme';
import ControlsSeparator from '.';

describe('Controls separator', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ControlsSeparator>
        <button type='button'>anything</button>
        <button type='button'>everything</button>
      </ControlsSeparator>
    );
  });

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('align right', () => {
    wrapper.setProps({ align: 'right' });
    expect(wrapper).toMatchSnapshot();
  });
});
