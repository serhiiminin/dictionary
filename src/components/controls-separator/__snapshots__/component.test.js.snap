// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls separator align right 1`] = `
<div
  className="undefined undefined"
>
  <button
    type="button"
  >
    anything
  </button>
  <button
    type="button"
  >
    anything
  </button>
</div>
`;

exports[`Controls separator render 1`] = `
<div
  className="undefined undefined"
>
  <button
    type="button"
  >
    anything
  </button>
  <button
    type="button"
  >
    anything
  </button>
</div>
`;
